smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/SmithyModelFileInfoGenerator.kt

```diff
@@ -0,0 +1,25 @@
+package software.amazon.smithy.swift.codegen
+
+import software.amazon.smithy.aws.traits.ServiceTrait
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.model.getTrait
+
```

Comment:
This just generates a very simple JSON file named `smithy-model-info.json` into the directory where the Swift source files go.  Its purpose is to tell the Swift code generator where it can find the JSON model file for this service client, relative to project root. Example contents:
```json
{"path":"codegen/sdk-codegen/aws-models/s3.json"}
```